src/util/codec/mysql/json/json_array.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME: remove following later
+#![allow(dead_code)]
+
+use super::Json;
+use super::super::super::datum::Datum;
+use super::super::super::Result;
+
+impl Json {
+    // https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-array
+    pub fn to_array(elems: &[Datum]) -> Result<Json> {
```

Comment:
I prefer to implement this function in `evaluator` with the following reasons:
1. we never use `Datum` in this mod
2. keep consistent with `tidb`